packages/ui-kit/src/Tooltip/Tooltip/Tooltip.tsx

```diff
@@ -41,24 +53,101 @@ export function Tooltip({
         return;
       }
 
-      onClose && onClose(e);
+      onClose?.(e);
     },
     [hoverBehavior, onOuterClick, onClose],
   );
 
+  const arrowRef = useRef<HTMLDivElement | null>(null);
+
+  const placement = useMemo(
+    () => calculatePlacement(getPlacement, placementProp, position) || defaultPlacement,
+    [getPlacement, placementProp, position],
+  );
+
+  const {
+    refs,
+    x: floatingX,
+    y: floatingY,
+    strategy,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+    placement: finalPlacement,
+  } = useFloating({
+    placement: placement,
+    open,
+    whileElementsMounted: autoUpdate,
+    middleware: [
+      offset(offsetValue),
+      flip({
+        fallbackAxisSideDirection: 'start',
+      }),
+      shift({ padding: 5 }),
+      arrow({ element: arrowRef }),
+    ],
+  });
+
+  useLayoutEffect(() => {
+    if (triggerRef) refs.setReference(triggerRef.current);
+  }, [triggerRef, refs]);
+
+  useLayoutEffect(() => {
+    if (position)
+      refs.setReference({
+        getBoundingClientRect() {
+          return {
+            width: 0,
+            height: 0,
+            x: position.x,
+            y: position.y,
+            top: position.y,
+            left: position.x,
+            right: position.x,
+            bottom: position.y,
+          };
+        },
+      });
+  }, [position, refs]);
+
+  const x = floatingX ?? 0;
+  const y = floatingY ?? 0;
+
+  const positionVariables = useMemo<CSSProperties>(
+    () =>
+      ({
+        '--tooltip-arrox-x-position': arrowX != null ? `${arrowX}px` : '',
+        '--tooltip-arrow-y-position': arrowY != null ? `${arrowY}px` : '',
+        '--tooltip-x-position': `${x}px`,
+        '--tooltip-y-position': `${y}px`,
+        '--tooltip-placement': strategy,
+      } as CSSProperties),
+    [arrowX, arrowY, strategy, x, y],
+  );
+
+  const arrowSide = useMemo(() => {
+    const side = finalPlacement.split('-')[0] || defaultPlacement;
+    return `arrow-${side}`;
+  }, [finalPlacement]);
+
+  if (!open) return null;
+
+  if (!triggerRef && !position) throw new Error('Tooltip: either triggerRef or position should be provided.');
+  if (!!position && !!triggerRef) throw new Error('Tooltip: only one of triggerRef or position should be provided.');
```

Comment:
if we must have one of position or triggerRef maybe it's a good idea to have one required prop for them that would be type of triggerRef | position, what do you think?

Reply:
@propakov Good idea I created single `anchor` prop with union type

Reply:
great name selection!